src/constructs/VPC.ts

```diff
@@ -0,0 +1,52 @@
+import { SecurityGroup, SubnetType, Vpc } from "@aws-cdk/aws-ec2";
+import { Construct as CdkConstruct } from "@aws-cdk/core";
+import { FromSchema } from "json-schema-to-ts";
+import { AwsCfInstruction } from "@serverless/typescript";
+import { AwsConstruct, AwsProvider } from "../classes";
+
+const VPC_DEFINITION = {
+    type: "object",
+    properties: {
+        type: { const: "vpc" },
+    },
+    additionalProperties: false,
+    required: [],
+} as const;
+
+type Configuration = FromSchema<typeof VPC_DEFINITION>;
+
+export class VPC extends AwsConstruct {
+    public static type = "vpc";
+    public static schema = VPC_DEFINITION;
+
+    private readonly vpc: Vpc;
+
+    constructor(scope: CdkConstruct, id: string, configuration: Configuration, private provider: AwsProvider) {
+        super(scope, id);
+
+        this.vpc = new Vpc(this, "VPC", {
+            maxAzs: 2,
+        });
+
+        const privateSubnets = this.vpc.selectSubnets({
+            subnetType: SubnetType.PRIVATE,
+        }).subnetIds;
+
+        const lambdaSecurityGroup = new SecurityGroup(this, "AppSecurityGroup", {
```

Comment:
I think we need to add an egress here to allow outbound trafic on http/https

Reply:
Yes, I would even say allow all outbound traffic.

Reply:
Done in the latest commit